src/measure_theory/measure/haar.lean

```diff
@@ -574,43 +574,54 @@ additive group."]
 def haar [locally_compact_space G] : measure G :=
 haar_measure $ classical.choice (topological_space.nonempty_positive_compacts G)
 
-section unique
+section second_countable
 
 variables [second_countable_topology G]
 
-/-- The Haar measure is unique up to scaling. More precisely: every σ-finite left invariant measure
-  is a scalar multiple of the Haar measure. -/
+/-- A variant on `haar_measure_unique` where we only require that `μ` is finite on `K₀`
+  (instead of all compact sets). -/
 @[to_additive]
-theorem haar_measure_unique (μ : measure G) [sigma_finite μ] [is_mul_left_invariant μ]
-  (K₀ : positive_compacts G) : μ = μ K₀.1 • haar_measure K₀ :=
+theorem haar_measure_unique' (μ : measure G) [sigma_finite μ] [is_mul_left_invariant μ]
+  (K₀ : positive_compacts G) (hμK₀ : μ K₀.1 < ∞) : μ = μ K₀.1 • haar_measure K₀ :=
 begin
-  ext1 s hs,
-  have := measure_mul_measure_eq μ (haar_measure K₀) K₀.2.1 hs,
-  rw [haar_measure_self, one_mul] at this,
-  rw [← this (by norm_num), smul_apply],
+  refine (measure_eq_div_smul μ (haar_measure K₀) K₀.2.1.measurable_set
+    (measure_pos_of_nonempty_interior _ K₀.2.2).ne' K₀.2.1.measure_lt_top hμK₀).trans _,
+  rw [haar_measure_self, ennreal.div_one]
 end
 
+/-- The Haar measure is unique up to scaling. More precisely: every σ-finite left invariant measure
+  finite on open compacts is a scalar multiple of the Haar measure.
+  This is slightly weaker than assuming that `μ` is a Haar measure (in particular we don't require
+  `μ ≠ 0`). -/
+@[to_additive]
+theorem haar_measure_unique (μ : measure G) [sigma_finite μ] [is_mul_left_invariant μ]
+  [is_finite_measure_on_compacts μ] (K₀ : positive_compacts G) : μ = μ K₀.1 • haar_measure K₀ :=
```

Comment:
This looks weaker than the previous version: there is a new `[is_finite_measure_on_compacts μ]` hypothesis, that was not present before. Is that right, or am I missing something?

Reply:
Does the docstring of `haar_measure_unique'` answer your question?
I think this special case is useful, because in most cases we will have `is_finite_measure_on_compacts`, and in that case we have 1 fewer explicit argument.

Reply:
Sorry but no, I don't see an answer to my question in that docstring. What I don't understand is that according to that diff, we previously had (line 584)
```lean
theorem haar_measure_unique (μ : measure G) [sigma_finite μ] [is_mul_left_invariant μ]
  (K₀ : positive_compacts G) : μ = μ K₀.1 • haar_measure K₀ :=
```
Now you replace it with the two theorems
```lean
theorem haar_measure_unique' (μ : measure G) [sigma_finite μ] [is_mul_left_invariant μ]
  (K₀ : positive_compacts G) (hμK₀ : μ K₀.1 ≠ ∞) : μ = μ K₀.1 • haar_measure K₀ :=

theorem haar_measure_unique (μ : measure G) [sigma_finite μ] [is_mul_left_invariant μ]
  [is_finite_measure_on_compacts μ] (K₀ : positive_compacts G) : μ = μ K₀.1 • haar_measure K₀ :=
```
Both look weaker than the original because they both have additional hypotheses. Why not keep the theorem in its original version?